### PR TITLE
[eclipse/xtext#1224] Detect local Jenkins environment

### PR DESCRIPTION
**Pull request by kthoms**
Build steps defined in Jenkinsfile pass the built-in environment
variable 'JENKINS_URL' to the Gradle/Maven executions. This is evaluated
in the build scripts for upstream repository URLs. On Xtext JIPP this
will use upstream repos from JIPP. In local builds outside of Jenkins
the property defaults to Typefox CI like before.

Extend function jenkinsPipelineRepo() by upstreamBranch parameter and use this in the call.
Use jenkinsPipelineRepo() also for lsp4j.

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: gradle/bootstrap-setup.gradle [hunk 1]
```diff
@@ -1,13 +1,16 @@
 /*
  * Root project configuration that is reused by subprojects to apply the Xtend compiler.
  */
+if (!hasProperty('JENKINS_URL')) {
```
Comment (cdietrich) [inline gradle/bootstrap-setup.gradle]: is hasProperty = -P or -D i am confused now

Comment (kthoms) [reply gradle/bootstrap-setup.gradle]: In Gradle it is "-P". In Maven "-D".

Comment (cdietrich) [reply gradle/bootstrap-setup.gradle]: :+1:

**Comment by kthoms** [comment]
The JENKINS_URL all-caps is a match to the build-in environment variable in Jenkins, see https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-belowJenkinsSetEnvironmentVariables

> So gradle uses -P to pass properties whereas we have to use -D in mvn land?

Right. Gradle uses project properties with -P (https://docs.gradle.org/current/userguide/command_line_interface.html#sec:command_line_completion). Maven uses system properties with -D. Gradle also knows -D system properties, could be that this also works. But I think the right way is -P here.

**Comment by kthoms** [comment]
Updated. URL for lsp4j stays hard-coded.

**Pull request merged**